design-proposals/klip-55-tablespaces.md

```diff
@@ -1,3 +1,210 @@
+    ...
+);
+```
+
+The default tablespace is `ksql_default`.
+
+**Creating a tablespace**
+
+You can supply one or more key/value parameters. These parameters are server-level
+configuration parameters that you want to group together.
+
+```sql
+CREATE TABLESPACE tablespace_name WITH (
+    property_name = expression [,
+    ...
+    ]
+);
+```
+
+By default, no objects are placed within a new tablespace.
+
+- TODO: what happens if `tablespace_name` already exists?
+
+**Adding a newly created object to a tablespace**
+
+Explicitly define what tablespace a stream or table should reside in.
+
+```sql
+CREATE [STREAM|TABLE] obj WITH (
+    'tablespace' = 'tablespace_name',
+    ...
+);
+```
+
+If `tablespace_name` doesn't exist, `obj`'s creation is rejected with a clear error message why.
+
+**Seeing what tablespace an object is in**
+
+```sql
+DESCRIBE obj EXTENDED;
+```
+
+Output fragment:
+
+```
+Name                 : KSQL_PROCESSING_LOG
+Type                 : STREAM
+Timestamp field      : Not set - using <ROWTIME>
+Tablespace           : tablespace_name
+```
+
+**Setting a default tablespace**
+
+You can override the default tablespace so you don't
+need to supply it to every proceeding object that gets created.
+
+```sql
+SET 'ksql.default.tablespace' = 'tablespace_name';
+```
+
+If `tablespace_name` doesn't exist, `obj`'s creation is rejected with a clear error message why.
+
+**Seeing the current default tablespace**
+
+```sql
+SHOW PROPERTIES;
+```
+
+TODO: can we get more granular than that?
+
+**Changing an existing object's tablespace**
+
+```sql
+ALTER <STREAM|TABLE> obj_name SET 'tablespace' = 'tablespace_name';
+```
+
+**Changing a tablespace's configuration**
+
+Causes all objects in the tablespace to be immediately reconfigured
+with the new parameters.
+
+```sql
+ALTER TABLESPACE tablespace_name SET 'key' = 'value';
+```
+
+TBD: should this support bulk updating parameters?
+
+**Viewing a tablespace's configuration**
+
+```sql
+DESCRIBE tablespace_name [EXTENDED];
+```
+
+Output:
+
+```sql
++-----------+--------+
+| parameter | value  |
++-----------+--------+
+|       k1  | v1     |
+|       k2  | v2     |
+|       k3  | v3     |
++-----------+--------+
+```
+
+**Dropping a tablespace**
+
+`DROP` only succeeds if it has no objects in it. If it has objects, it lists
+out which ones need to be dropped first.
+
+```sql
+DROP TABLESPACE [IF EXISTS] tablespace_name;
+```
+
+**Seeing all objects in a tablespace**
+
+TBD.
+
+**Precedence**
+
+Parameters take the following precedence:
+
+1. Query-level configuration
+2. Tablespace configuration
+3. Server configuration
+
+## Design
+
+TBD.
+
+## Test plan
+
+TBD.
+
+## LOEs and Delivery Milestones
+
+TBD.
+
+## Documentation Updates
+
+- Add a new how-to guide
+- Update the language reference section with the new grammar
+
+## Compatibility Implications
+
+All existing objects get placed into the default tablespace.
+
+## Security Implications
+
+TBD.
+
+## Open questions
+
```

Comment:
i didn't see a discussion of what happens with non-persistent queries ? do they also execute in the context of the current space ?